eth/protocols/snap/handler.go

```diff
@@ -281,7 +284,12 @@ func ServiceGetAccountRangeQuery(chain *core.BlockChain, req *GetAccountRangePac
 	if err != nil {
 		return nil, nil
 	}
-	it, err := chain.Snapshots().AccountIterator(req.Root, req.Origin)
+	var it snapshot.AccountIterator // ugly hack
+	if chain.TrieDB().Scheme() == rawdb.HashScheme {
+		it, err = chain.Snapshots().AccountIterator(req.Root, req.Origin)
+	} else {
+		it, err = chain.TrieDB().AccountIterator(req.Root, req.Origin)
```

Comment:
I don't understand this. Why are you using triedb account iterator in pbss-mode? (also below)

Reply:
In the path mode, the legacy state snapshot is not existent anymore. The original state snapshot data will be over taken by the pathdb (the states inside of pathdb).

Therefore, constructing the iterator from pathdb is the only efficient/equivalent way to serve the snap state requests